packages/ibm-products/src/components/Datagrid/Datagrid/addons/CustomizeColumns/DraggableItemsList.js

```diff
@@ -13,123 +13,219 @@ import DraggableElement from '../../DraggableElement';
 import { pkg } from '../../../../../settings';
 import getColTitle from '../../../utils/getColTitle';
 
+import {
+  DndContext,
+  KeyboardSensor,
+  PointerSensor,
+  closestCenter,
+  useSensor,
+  useSensors,
+} from '@dnd-kit/core';
+import {
+  SortableContext,
+  verticalListSortingStrategy,
+} from '@dnd-kit/sortable';
+
 const blockClass = `${pkg.prefix}--datagrid`;
 
 export const DraggableItemsList = ({
   columns,
   filterString,
-  focusIndex,
-  getNextIndex,
-  // moveElement,
+  id,
+  moveElement,
   onSelectColumn,
   setAriaRegionText,
-  setColumnsObject,
-  setFocusIndex,
 }) => {
+  const draggableClass = `${blockClass}__draggable-item`;
+  const visibleCols = columns
+    // hide the columns without Header, e.g the sticky actions, spacer
+    .filter((colDef) => {
+      return !!getColTitle(colDef);
+    })
+    .filter(Boolean)
+    .filter((colDef) => !colDef.isAction)
+    .filter((colDef) => {
+      return (
+        filterString.length === 0 ||
+        (getColTitle(colDef)?.toLowerCase().includes(filterString) &&
+          colDef.id !== 'spacer')
+      );
+    });
+
+  // let localRefCopy;
+  const handleDragEnd = (event) => {
+    const { active, over } = event;
+
+    const fromVisibleIndex = columns.findIndex((col) => col.id === active.id);
+    const toVisibleIndex = columns.findIndex((col) => col.id === over.id);
+    const colTitle = getColTitle(visibleCols[fromVisibleIndex]);
+
+    setAriaRegionText(
+      `${colTitle} dropped. New position ${toVisibleIndex + 1} of ${
+        visibleCols.length
+      }.`
+    );
+
+    const fromIndex = columns.findIndex((col) => col.id === active.id);
+    const toIndex = columns.findIndex((col) => col.id === over.id);
+
+    moveElement(fromIndex, toIndex);
+  };
+
+  const handleDragStart = (event) => {
+    const { active } = event;
+
+    const fromIndex = visibleCols.findIndex((col) => col.id === active.id);
+    const colTitle = getColTitle(visibleCols[fromIndex]);
+
+    setAriaRegionText(
+      `${colTitle} grabbed. Current position ${fromIndex + 1} of ${
+        visibleCols.length
+      }.`
+    );
+  };
+
+  const handleDragUpdate = (event) => {
+    const { active, over } = event;
+
+    const fromIndex = visibleCols.findIndex((col) => col.id === active.id);
```

Comment:
I believe it is one of these two lines that is causing an error to be thrown if dragging from the checkbox